src/coreclr/jit/importercalls.cpp

```diff
@@ -3364,6 +3366,28 @@ GenTree* Compiler::impIntrinsic(GenTree*                newobjThis,
                                 retNode = gtNewFalse();
                             }
                             break;
+                        case NI_System_Type_get_IsGenericType:
+                        {
+                            // If the type is __Canon, we just give up on expanding the intrinsic
+                            if (info.compCompHnd->getClassAttribs(hClass) & CORINFO_FLG_SHAREDINST)
```

Comment:
Note that this is going to give up on `List<__Canon>` as well.

Reply:
Ah, yup @EgorBo also just mentioned that, it wasn't entirely clear to me that the flag meant "any part of this type is `__Canon`", I thought it just meant "the type itself is actually `__Canon`". Will go fix that 😅